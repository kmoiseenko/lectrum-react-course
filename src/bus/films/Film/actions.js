// Types
import { types } from './types';

import { api } from './../../../api';

export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    filmFetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());
        try {
            const response = await api.film.fetch(id);

            if (response.status === 200) {
                try {
                    const result = await response.json();

                    dispatch(filmActions.fill(result));
                } catch(error) {
                    console.log(error);
                }
            } else {
                const error = {
                    status: response.status
                };

                dispatch(filmActions.setFetchingError(error));
            }

            dispatch(filmActions.stopFetching());
        } catch(error) {
            console.log(error)
        }
    }
});