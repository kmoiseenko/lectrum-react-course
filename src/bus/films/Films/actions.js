// Types
import { types } from './types';

import { api } from './../../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    filmsFetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());
        try {
            const response = await api.films.fetch();

            if (response.status === 200) {
                try {
                    const { results } = await response.json();

                    dispatch(filmsActions.fill(results));
                } catch(error) {
                    console.log(error);
                }
            } else {
                const error = {
                    status: response.status
                };

                dispatch(filmsActions.setFetchingError(error));
            }

            dispatch(filmsActions.stopFetching());
        } catch(error) {
            console.log(error);
        }
    }
});