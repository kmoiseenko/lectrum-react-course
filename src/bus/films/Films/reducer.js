// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const filmsReducer = (state = initialState, { type, payload }) => {
    switch (type) {
        case types.FILMS_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILMS_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILMS_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILMS_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
}