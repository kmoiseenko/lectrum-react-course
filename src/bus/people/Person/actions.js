// Types
import { types } from './types';

import { api } from './../../../api';

export const personActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.PERSON_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.PERSON_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.PERSON_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.PERSON_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    personFetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.PERSON_FETCH_ASYNC
        });

        dispatch(personActions.startFetching());
        const response = await api.person.fetch(id);
        
        if (response.status === 200) {
            const result = await response.json();

            dispatch(personActions.fill(result));
        } else {
            const error = {
                status: response.status
            };

            dispatch(personActions.setFetchingError(error));
        }

        dispatch(personActions.stopFetching());
    }
});