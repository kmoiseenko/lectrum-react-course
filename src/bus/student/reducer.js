// Types
import { types } from './types';

const initialState = {
    firstName: '',
    surname: '',
    age: '',
    email: '',
    sex: '',
    speciality: '',
    password: '',
    confirmPassword: ''
};

export const studentReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.SET_STUDENT:
            return { ...state, ...payload };

        default:
            return state;
    }
}