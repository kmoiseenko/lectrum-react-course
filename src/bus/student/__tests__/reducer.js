import { studentReducer } from "./../reducer";
import { studentActions } from "./../actions";
import { types } from "./../types";

describe("Student reducer:", () => {
    test("Should return initial state by default", () => {
        expect(studentReducer(void 0, {})).toMatchSnapshot();
    });

    test(`Should handle ${types.SET_STUDENT} action`, () => {
        expect(studentReducer(void 0, studentActions.setStudent())).toMatchSnapshot();
    });
});