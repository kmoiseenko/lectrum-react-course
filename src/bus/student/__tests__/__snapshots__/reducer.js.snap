// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student reducer: Should handle SET_STUDENT action 1`] = `
Object {
  "age": "",
  "confirmPassword": "",
  "email": "",
  "firstName": "",
  "password": "",
  "sex": "",
  "speciality": "",
  "surname": "",
}
`;

exports[`Student reducer: Should return initial state by default 1`] = `
Object {
  "age": "",
  "confirmPassword": "",
  "email": "",
  "firstName": "",
  "password": "",
  "sex": "",
  "speciality": "",
  "surname": "",
}
`;
